control-agent/text-to-speech/speech.py

```diff
@@ -145,14 +159,99 @@ def recognize(blobData):
         conn.close()
     except Exception as e:
         print("Vision Error: ", e)
+        return "error"
+
+    return json.loads(data)
+
+
+def recognize_face(blobData):
+    """Use Azure computer vision recognize from an image as bytes
+    Keyword arguments:
+    blobData -- bytes data of an image
+    """
+
+    key, url = getFaceKeys()
+
+    headers = {
+        # Request headers
+        "Content-Type": "application/octet-stream",
+        "Ocp-Apim-Subscription-Key": key,
+    }
+
+    params = urllib.parse.urlencode(
+        {
+            # Request parameters
+            "detectionModel": "detection_03",
+            "returnFaceId": "true",
+            "returnFaceLandmarks": "false",
+        }
+    )
+
+    body = blobData
+
+    try:
+        conn = http.client.HTTPSConnection(url)
+        conn.request("POST", "/face/v1.0/detect?%s" % params, body, headers)
+        response = conn.getresponse()
+        data = response.read()
+        print(data)
+        conn.close()
+    except Exception as e:
+        print("Vision Error: ", e)
+        return "error"
 
     return json.loads(data)
 
 
-<<<<<<< HEAD
-# Tests
-# result = recognize("https://robots.ieee.org/robots/stretch/stretch-1200x630.jpg")
-=======
+def identify_face(face_id):
+
+    key, url = getFaceKeys()
+
+    headers = {
+        # Request headers`
+        "Content-Type": "application/json",
+        "Ocp-Apim-Subscription-Key": key,
+    }
+
+    params = urllib.parse.urlencode(
+        {
+            # Request parameters
+            "detectionModel": "detection_03",
+            "returnFaceId": "true",
+            "returnFaceLandmarks": "false",
+        }
+    )
+
+    body = {
+        "PersonGroupId": "53cdefb1-c211-4215-a944-6b64128fa39f",
+        "faceIds": [str(face_id)],
+        "maxNumOfCandidatesReturned": 1,
+        "confidenceThreshold": 0.5,
+    }
+
+    try:
+        conn = http.client.HTTPSConnection(url)
+        conn.request("POST", "/face/v1.0/identify?%s" % params, str(body), headers)
+        response = conn.getresponse()
+        data = response.read()
+        print(data)
+        conn.close()
+    except Exception as e:
+        print("Vision Error: ", e)
+        return "error"
+
+    # face logic
+    # TODO remove hard coding
+    results = json.loads(data)
+    if (
+        results[0]["candidates"][0]["personId"]
+        == "f9611b03-dc30-48bd-88f5-9ce251f688b6"
```

Comment:
Put this in a variable with a single leading underscore whilst it's hard coded -  https://pep8.org/